cmd/ceremony/cert.go

```diff
@@ -321,31 +320,23 @@ func (fr *failReader) Read([]byte) (int, error) {
 }
 
 func generateCSR(profile *certProfile, randReader *hsmRandReader, pubBytes []byte, pub crypto.PublicKey, signer crypto.Signer) ([]byte, error) {
-	// currently Go doesn't support all of the convenience fields for x509.CertificateRequest
-	// that x509.Certificate has. Instead of doing all of the manual extension construction
-	// ourselves here we just create a throwaway self-signed certificate and then dump all
-	// of the generated extensions into a x509.CertificateRequest. In the future Go should
-	// support doing this properly itself, but for now this is the easiest approach.
+	// Go doesn't appear likely to adopt any convenience fields in
+	// x509.CertificateRequest that x509.Certificate has and in go 1.16 .
+	// We do not intend to use CSRs in signing ceremonies, but have this
+	// function in case a CSR is necessary for a cross signature from a CA
+	// that requires a CSR. In that case, the signing party will need to
+	// sign with appropriate extensions, but we will provide a CSR with
+	// enough information for the signing party to generate an appropriate
+	// certificate.
```

Comment:
There's a temptation when editing code that has a comment to always update that comment rather than deleting it. Comments are good, right? But in this case we're removing complexity so we can remove the comment. Leaving the comment but removing the complexity would actually _increase_ the confusion of someone reading the code because it attempts to describe a state of the world that no longer exists.